imports/table/shared.lua

```diff
@@ -9,47 +9,54 @@ local pairs = pairs
 ---@return boolean
 ---Checks if tbl contains the given values. Only intended for simple values and unnested tables.
 local function contains(tbl, value)
-	if type(value) ~= 'table' then
-		for _, v in pairs(tbl) do
-			if v == value then return true end
-		end
-	else
-		local matched_values = 0
-		local values = 0
-		for _, v1 in pairs(value) do
-			values += 1
-
-			for _, v2 in pairs(tbl) do
-				if v1 == v2 then matched_values += 1 end
-			end
-		end
-		if matched_values == values then return true end
-	end
+    if not next(tbl) then return false end
+
+    if type(value) ~= 'table' then
+        for _, v in pairs(tbl) do
+            if v == value then 
+                return true 
+            end
+        end
+        
+        return false
+    else
+        local set = {}
+        for _, v in pairs(tbl) do
+            set[v] = true
+        end
+
+        for _, v in pairs(value) do
+            if not set[v] then 
+                return false
+            end
+        end
 
-	return false
+        return true
+    end
 end
 
 ---@param t1 any
 ---@param t2 any
 ---@return boolean
 ---Compares if two values are equal, iterating over tables and matching both keys and values.
 local function table_matches(t1, t2)
-	local type1, type2 = type(t1), type(t2)
-
-	if type1 ~= type2 then return false end
-	if type1 ~= 'table' and type2 ~= 'table' then return t1 == t2 end
+    if type(t1) ~= type(t2) then return false end
+    if type(t1) ~= 'table' then return t1 == t2 end
```

Comment:
Considering we're micro-optimising here, there's no reason not to reuse type1.

Reply:
While we're at it we can get a small boost by comparing table types and lengths.

```lua
    local tabletype1 = table.type(t1)

    if tabletype1 ~= table.type(t2) or (tabletype1 == 'array' and #t1 ~= #t2) then
        return false
    end
```